arch/riscv/riscv_features.c

```diff
@@ -14,7 +14,10 @@
 
 int Z_INTERNAL is_kernel_version_greater_or_equal_to_6_5() {
     struct utsname buffer;
-    uname(&buffer);
+    if (uname(&buffer) == -1) {
+        // uname failed
+        return 0;
+    }
```

Comment:
Only way this can fail is if the host name is too long, usually longer than 64 characters reserved for each of the fields in the struct. Even then some versions of `uname` function will just truncate the host name and leave NULL-termination out. However if `-1` is returned, the other fields are guaranteed to be uninitialised.

Not ignoring the return value makes it clear that the call to `uname` and `buffer` in stack can't be optimised out, which would cause the contents of `buffer` later below to be uninitialised. Some compilers will issue warning if the contents of struct member can be uninitialised (not guaranteed to be zero-filled), but that might require enabling "higher" warning level or static analysis.